larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewThreeDKinkTool.h

```diff
@@ -0,0 +1,172 @@
+/**
+ *  @file   larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewThreeDKinkTool.h
+ *
+ *  @brief  Header file for the two view three d kink base tool
+ *
+ *  $Log: $
+ */
+#ifndef TWO_VIEW_THREE_D_KINK_TOOL_H
+#define TWO_VIEW_THREE_D_KINK_TOOL_H 1
+
+#include "larpandoracontent/LArObjects/LArPointingCluster.h"
+
+#include "larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewTransverseTracksAlgorithm.h"
+
+namespace lar_content
+{
+
+/**
+ *  @brief  TwoViewThreeDKinkTool class
+ */
+class TwoViewThreeDKinkTool : public TransverseMatrixTool
+{
+public:
+    /**
+     *  @brief  Constructor
+     *
+     *  @param  nCommonClusters the number of common clusters to select
+     */
+    TwoViewThreeDKinkTool();
+
+    /**
+     *  @brief  Destructor
+     */
+    ~TwoViewThreeDKinkTool();
```

Comment:
Declare as virtual.

Reply:
Does this need to be declared virtual, now that TwoViewThreeDKinkTool has no derived classes in the current implementation?

Reply:
Yes, you still want it, as it's always possible for future updates to inherit from this class, and this class already inherits from another. In general, all destructors should be tagged as virtual, this ensures that the correct destructor is called on an object however it is being accessed at any given time. If you don't declare the destructor virtual you can end up in situations where a base class destructor gets called instead and so the derived object may not be completely destroyed.